resources/vaft-ublock-origin.js

```diff
@@ -0,0 +1,865 @@
+        document.addEventListener('mozvisibilitychange', block, true);
+        document.addEventListener('hasFocus', block, true);
+        if (/Firefox/.test(navigator.userAgent)) {
+            Object.defineProperty(document, 'mozHidden', {
+                get() {
+                    return false;
+                }
+            });
+        } else {
+            Object.defineProperty(document, 'webkitHidden', {
+                get() {
+                    return false;
+                }
+            });
+        }
+    } catch (err) {}
+    //Send settings updates to worker.
+    window.addEventListener('message', (event) => {
+        if (event.source != window) {
+            return;
+        }
+        if (event.data.type && event.data.type == 'SetTwitchAdblockSettings' && event.data.settings) {
+            TwitchAdblockSettings = event.data.settings;
+        }
+    }, false);
+    function declareOptions(scope) {
+        scope.AdSignifier = 'stitched';
+        scope.ClientID = 'kimne78kx3ncx6brgo4mv6wki5h1ko';
+        scope.ClientVersion = 'null';
+        scope.ClientSession = 'null';
+        scope.PlayerType2 = 'embed'; //Source
+        scope.PlayerType3 = 'site'; //Source
+        scope.PlayerType4 = 'autoplay'; //360p
+        scope.CurrentChannelName = null;
+        scope.UsherParams = null;
+        scope.WasShowingAd = false;
+        scope.GQLDeviceID = null;
+        scope.IsSquadStream = false;
+        scope.StreamInfos = [];
+        scope.StreamInfosByUrl = [];
+        scope.MainUrlByUrl = [];
+        scope.EncodingCacheTimeout = 60000;
+        scope.ClientIntegrityHeader = null;
+        scope.AuthorizationHeader = null;
+    }
+    declareOptions(window);
+    var TwitchAdblockSettings = {
+        BannerVisible: true,
+        ForcedQuality: null,
+    };
+    var twitchWorkers = [];
+    var adBlockDiv = null;
+    var OriginalVideoPlayerQuality = null;
+    var IsPlayerAutoQuality = null;
+    const oldWorker = window.Worker;
+    window.Worker = class Worker extends oldWorker {
+        constructor(twitchBlobUrl) {
+            var jsURL = getWasmWorkerUrl(twitchBlobUrl);
+            if (typeof jsURL !== 'string') {
+                super(twitchBlobUrl);
+                return;
+            }
+            var newBlobStr = `
+                ${getStreamUrlForResolution.toString()}
+                ${getStreamForResolution.toString()}
+                ${stripUnusedParams.toString()}
+                ${processM3U8.toString()}
+                ${hookWorkerFetch.toString()}
+                ${declareOptions.toString()}
+                ${getAccessToken.toString()}
+                ${gqlRequest.toString()}
+                ${adRecordgqlPacket.toString()}
+                ${tryNotifyTwitch.toString()}
+                ${parseAttributes.toString()}
+                declareOptions(self);
+                self.TwitchAdblockSettings = ${JSON.stringify(TwitchAdblockSettings)};
```

Comment:
Eeek, this is scary

Reply:
Oh, I see the first 11 are just printing the content of functions. Line 104 is printing the JSON which might have been problematic but we aren't in an HTML context.

Reply:
Removed `TwitchAdblockSettings` and related code in https://github.com/pixeltris/TwitchAdSolutions/commit/02447140c7522721601ac62c4fb47c7757e054d7 as it wasn't used.

And yes the weird looking code above is injecting code into their `Worker` to intercept `fetch` to modify the result of m3u8 requests.